### PR TITLE
Load Subscriptions Core and WCPay Subscriptions support classes behind a feature flag

### LINKED ISSUE #2987: Add a feature flag behind loading WCPay Subscriptions (including subscriptions core)
### Description

As part of our scope call this week, we decided on doing a soft launch for WCPay Subscriptions (p2: pdjTHR-7e-p2) and only load subscriptions functionality based on the connected accounts country (i.e. US-only)

I had a bit of a look into how we could do this and here's one solution:
 - Add a new function to `WC_Payments_Features` class. Something like `is_wcpay_subscriptions_enabled()` and have it allow us to pass in WC_Payments_Account as an optional param.
 - This function will check if the wp_option `_wcpay_feature_subscriptions` is set and if so, return that value.
 - If it's not set and `WC_Payments_Account` is passed to the function, check if the account is in the US, and if so, set the `_wcpay_feature_subscriptions` option.
 - Inside [`WC_Payments::init()` we'll need to surround the code which loads WCPay subscription classes](https://github.com/Automattic/woocommerce-payments/blob/feature/stripe_billing/includes/class-wc-payments.php#L313-L315) with something like:

```
if ( WC_Payments_Features::is_wcpay_subscriptions_enabled( self::$account ) ) {
	include_once WCPAY_ABSPATH . '/includes/subscriptions/class-wc-payments-subscriptions.php';
	WC_Payments_Subscriptions::init( self::$api_client, self::$customer_service );
}
```
 - Make sure `subscriptions-core` isn't loaded by surrounding the [`require_once` and `new WC_Subscriptions_Core_Plugin();` inside `wcpay_init_subscriptions_core()`](https://github.com/Automattic/woocommerce-payments/blob/feature/stripe_billing/woocommerce-payments.php#L123-L125) with a similar condition to above:

```
if ( WC_Payments_Features::is_wcpay_subscriptions_enabled() ) {
	require_once $is_wcs_core_active ? WP_PLUGIN_DIR . '/woocommerce-subscriptions-core/' : WCPAY_SUBSCRIPTIONS_ABSPATH . 'includes/class-wc-subscriptions-core-plugin.php';
	new WC_Subscriptions_Core_Plugin();
}
```

At the time of initing core, we don't have access to the connected account so we'll only be checking for the feature flag option.

I think this is fine though. Once a site is connected the wp-option will be set and subscriptions-core will load correctly.

### PR DESCRIPTION
**Pull request by mattallan**
Fixes #2987 

#### Changes proposed in this Pull Request

With this PR, Subscriptions Core and WCPay Subscription will only be loaded when the feature flag `_wcpay_feature_subscriptions` is set in wp_options to `1`.

As part of our teams scope call last week, we discussed doing a soft-launch to only US based stores, however, as part of merging `feature/stripe_billing` into `develop` we decided to remove the US-only check and make it only check for the wp-option first.

~When I first added the new `WC_Payments_Features::is_wcpay_subscriptions_enabled()` function, I wrote it with the intention of having the WCPay Subscriptions US-only so I've commented out the code to be used later when we release this.~

**Chris update:** I have removed the commented out code, we can always add it back later when needed.

#### Testing instructions

1. Make sure you have Subscriptions extension deactivated
2. Checkout this `feature/stripe_billing`
3. Confirm WCPay Subscriptions is loaded (check **WooCommerce > Subscriptions**)
4. Checkout this branch
5. Without the feature flag set, confirm that no Subscriptions functionality exists (I just checked there's no Subscriptions settings or Subscriptions table)
6. Manually add `_wcpay_feature_subscriptions` in your wp_options table and set it to `1`
7. Confirm Subscriptions Core is loading (check Subscriptions Settings and table exists)
8. Set `_wcpay_feature_subscriptions` to `0` and check no subscriptions functionality is loaded (opposite of above checks)

-------------------

Something I'm interested in hearing others thoughts on is do we actually need surround the following code in a feature flag check:

```
include_once WCPAY_ABSPATH . '/includes/subscriptions/class-wc-payments-subscriptions.php';
WC_Payments_Subscriptions::init( self::$api_client, self::$customer_service );
```

On its own, I don't think most of those classes that get setup as part of this `init()` don't actually do anything.

Closes #2987

### CODE REVIEW

File: includes/class-wc-payments-features.php [hunk 1]
```diff
@@ -87,6 +89,27 @@ public static function is_account_overview_task_list_enabled() {
...
 	}
 
+	/**
+	 * Checks whether WCPay Subscriptions is enabled.
+	 *
+	 * @param WC_Payments_Account $wcpay_account Connected WCPay Account to check the location.
+	 * @return bool
+	 */
+	public static function is_wcpay_subscriptions_enabled( $wcpay_account = null ) {
+		$is_wcpay_subscriptions_enabled = get_option( self::WCPAY_SUBSCRIPTIONS_FLAG_NAME );
```
Comment (aprea) [inline includes/class-wc-payments-features.php:99]: **Nitpick:** Looks like the convention in this file is to set `'0'` as the default here and then perform a strict comparison against `'1'` in the return statement.

File: woocommerce-payments.php [hunk 2]
```diff
@@ -24,6 +24,7 @@
 define( 'WCPAY_SUBSCRIPTIONS_ABSPATH', __DIR__ . '/vendor/woocommerce/subscriptions-core/' );
 
 require_once __DIR__ . '/vendor/autoload_packages.php';
+require_once __DIR__ . '/includes/class-wc-payments-features.php';
```
Comment (aprea) [inline woocommerce-payments.php:27]: Moved this here from `includes/class-wc-payments.php` so we could access `WC_Payments_Features` a little earlier (see below)

**Comment by james-allan** [comment]
This looks good and works for me :)

**Review by james-allan — APPROVED** [review]

File: includes/class-wc-payments.php [hunk 3]
```diff
@@ -310,9 +309,11 @@ public static function init() {
 			new WC_Payments_Admin_Sections_Overwrite( self::get_account_service() );
 		}
 
-		// Load subscriptions (Stripe Billing).
-		include_once WCPAY_ABSPATH . '/includes/subscriptions/class-wc-payments-subscriptions.php';
-		WC_Payments_Subscriptions::init( self::$api_client, self::$customer_service );
+		// Load WCPay Subscriptions.
+		if ( WC_Payments_Features::is_wcpay_subscriptions_enabled() ) {
+			include_once WCPAY_ABSPATH . '/includes/subscriptions/class-wc-payments-subscriptions.php';
```
Comment (haszari) [inline includes/class-wc-payments.php:314]: @mattallan should we put a try/catch around this to gracefully fail when the subs core lib is not correctly in place? (I just hit this - I failed to include it in my test build)

**Pull request merged**